controllers/cruisecontroloperation_controller.go

```diff
@@ -0,0 +1,445 @@
+	log.V(1).Info("reconciling CruiseControlOperation custom resources")
+
+	ccOperationListClusterWide := banzaiv1alpha1.CruiseControlOperationList{}
+	err := r.DirectClient.List(ctx, &ccOperationListClusterWide, client.ListOption(client.InNamespace(request.Namespace)))
+	if err != nil {
+		return requeueWithError(log, err.Error(), err)
+	}
+
+	// Selecting the reconciling CruiseControlOperation
+	var currentCCoperation *banzaiv1alpha1.CruiseControlOperation
+	for i := range ccOperationListClusterWide.Items {
+		ccOperation := &ccOperationListClusterWide.Items[i]
+		if ccOperation.GetName() == request.Name && ccOperation.GetNamespace() == request.Namespace {
+			currentCCoperation = ccOperation
+		}
+	}
+
+	// Object is deleted...
+	if currentCCoperation == nil {
+		return reconciled()
+	}
+	// CruiseControlOperation has invalid operation type, reconciled
+	if !currentCCoperation.IsCurrentTaskOperationValid() {
+		log.Info("Koperator does not support this operation", "operation", currentCCoperation.GetCurrentTaskOp())
+		return reconciled()
+	}
+
+	kafkaClusterRef, err := getKafkaClusterReference(currentCCoperation)
+	if err != nil {
+		return requeueWithError(log, "couldn't get kafka cluster reference", err)
+	}
+
+	kafkaCluster := &banzaiv1beta1.KafkaCluster{}
+	err = r.Get(ctx, kafkaClusterRef, kafkaCluster)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			if !currentCCoperation.ObjectMeta.DeletionTimestamp.IsZero() {
+				log.Info("Cluster is gone already, remove finalizer")
+				controllerutil.RemoveFinalizer(currentCCoperation, cruiseControlOperationFinalizer)
+				if err := r.Update(ctx, currentCCoperation); err != nil {
+					return requeueWithError(log, "failed to remove finalizer from CruiseControlOperation", err)
+				}
+			}
+			return reconciled()
+		}
+		return requeueWithError(log, "failed to lookup referenced cluster", err)
+	}
+
+	//Adding finalizer
+	if err := r.addFinalizer(ctx, currentCCoperation); err != nil {
+		return requeueWithError(log, "finalizer could not be added", err)
+	}
+
+	// We only get scaler when we have not had mocked one for test
+	if _, ok := r.Scaler.(*scale.MockCruiseControlScaler); !ok {
+		r.Scaler, err = scale.NewCruiseControlScaler(ctx, scale.CruiseControlURLFromKafkaCluster(kafkaCluster))
+		if err != nil {
+			return requeueWithError(log, "failed to create Cruise Control Scaler instance", err)
+		}
+	} else {
+		// In tests we requeue faster
+		defaultRequeueIntervall = 1
+	}
+	// Checking Cruise Control availability
+	if !r.Scaler.IsUp() {
+		log.Info("requeue event as Cruise Control is not available (yet)")
+		return requeueAfter(defaultRequeueIntervall)
+	}
+
+	// Filtering out CruiseControlOperation by kafka cluster ref
+	var ccOperationsKafkaClusterFiltered []*banzaiv1alpha1.CruiseControlOperation
+	for i := range ccOperationListClusterWide.Items {
+		operation := &ccOperationListClusterWide.Items[i]
+		ret, err := getKafkaClusterReference(operation)
```

Comment:
I guess this variable is meant to be named as `ref` rather than "ret"?

Reply:
Ref is better, Thanks.